diff -ruN linux-5.15.130/drivers/usb/serial/option.c linux-5.15.130-org/drivers/usb/serial/option.c
--- linux-5.15.130/drivers/usb/serial/option.c	2023-09-16 16:26:56.547510858 +0800
+++ linux-5.15.130-org/drivers/usb/serial/option.c	2023-09-22 11:49:51.428201482 +0800
@@ -608,6 +608,28 @@
 /* TOZED LT70-C based on UNISOC SL8563 uses UNISOC's vendor ID */
 #define TOZED_PRODUCT_LT70C			0x4055
 
+/*add for meig modem supported {*/
+// #define MEIG_QCM_VENDOR_ID                      0x05C6
+// #define MEIG_QCM_PRODUCT_SLM750_SRM815_SLM868   0xF601
+
+#define MEIG_VENDOR_ID                          0x2DEE
+#define MEIG_PRODUCT_SLM790               		0x4D20
+#define MEIG_QCM_PRODUCT_NDIS              		0x4D22
+#define MEIG_QCM_PRODUCT_ECM                	0x4D23
+#define MEIG_QCM_PRODUCT_RNDIS                	0x4D38
+#define MEIG_QCM_PRODUCT_MBIM                	0x4D45
+
+#define MEIG_UNI_PRODUCT_ECM                	0x4D50
+#define MEIG_UNI_PRODUCT_RNDIS                	0x4D51
+#define MEIG_UNI_PRODUCT_NCM                	0x4D52
+#define MEIG_UNI_PRODUCT_MBIM                	0x4D53
+
+#define MEIG_ASR_PRODUCT_RNDIS                	0x4D57
+#define MEIG_ASR_PRODUCT_ECM                	0x4D58
+#define MEIG_ASR_PRODUCT_MBIM                	0x4D59
+
+/*}add for support meig end*/
+
 /* Device flags */
 
 /* Highest interface number which can be used with NCTRL() and RSVD() */
@@ -2264,6 +2286,26 @@
 	{ USB_DEVICE_AND_INTERFACE_INFO(SIERRA_VENDOR_ID, SIERRA_PRODUCT_EM9191, 0xff, 0xff, 0x30) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(SIERRA_VENDOR_ID, SIERRA_PRODUCT_EM9191, 0xff, 0, 0) },
 	{ USB_DEVICE_AND_INTERFACE_INFO(UNISOC_VENDOR_ID, TOZED_PRODUCT_LT70C, 0xff, 0, 0) },
+
+	/*add for meig modem supported {*/
+
+	// { USB_DEVICE(MEIG_QCM_VENDOR_ID,  MEIG_QCM_PRODUCT_SLM750_SRM815_SLM868),.driver_info = RSVD(4) | RSVD(5)| RSVD(6) | RSVD(7)|RSVD(8)},
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_PRODUCT_SLM790),.driver_info = RSVD(0) | RSVD(5)| RSVD(6) | RSVD(7) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_QCM_PRODUCT_NDIS), .driver_info = RSVD(4) | RSVD(5) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_QCM_PRODUCT_ECM),.driver_info = RSVD(4) | RSVD(5) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_QCM_PRODUCT_RNDIS) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_QCM_PRODUCT_MBIM) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_UNI_PRODUCT_ECM) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_UNI_PRODUCT_RNDIS),.driver_info = RSVD(0) | RSVD(1) | RSVD(7)| RSVD(8) | RSVD(9)},
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_UNI_PRODUCT_NCM),.driver_info = RSVD(0) | RSVD(1) | RSVD(7)| RSVD(8) | RSVD(9)},
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_UNI_PRODUCT_MBIM) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_ASR_PRODUCT_RNDIS) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_ASR_PRODUCT_ECM) },
+	{ USB_DEVICE(MEIG_VENDOR_ID,  MEIG_ASR_PRODUCT_MBIM) },
+
+
+
+	/*}add for support meig end*/
 	{ } /* Terminating entry */
 };
 MODULE_DEVICE_TABLE(usb, option_ids);
